linera-execution/src/evm/database.rs

```diff
@@ -378,4 +379,35 @@ where
             blob_excess_gas_and_price,
         })
     }
+
+    pub fn constructor_argument(&self) -> Result<Vec<u8>, ExecutionError> {
+        let mut runtime = self.runtime.lock().expect("The lock should be possible");
+        let constructor_argument = runtime.application_parameters()?;
+        Ok(serde_json::from_slice::<Vec<u8>>(&constructor_argument)?)
```

Comment:
Why do we use `serde_json` here? I imagine that in order for this to work, we'd have to serialize with serde as well.

Reply:
Because the application parameters are serialized with `serde_json`.